File: trap.py
#!/usr/bin/env python

from fractions import Fraction
import math

def trapezium(f,x,h):
    return (f(x) + f(x+h))/2.0

def integrate(f, a, b, steps, method):
    h = (b-a) / steps
    ival = h * sum(method(f, a+i*h, h) for i in range(steps))
    return ival

def f1(x):
    return 2*x

print integrate(f1, 1, math.pi, 100, trapezium)

File: javascript/trap.js
//----------------------------------------------
//
// trapezoidal integration in javascript.
//
// currently borked. 
//
//----------------------------------------------
function sum(n) {
    var v = function (x) {
        return sum (n + x);
    };

    v.valueOf = v.toString = function () {
        return n;
    };
    return v;
}

function trapezium(f, x, h) {
    return (f(x) + f(x+h))/2.0;
}

function f(x) {
    return 2*x;
}

function integrate(f, a, b, steps, method) {
    var h = (b-a) / steps;
    WScript.Echo(steps); 
    for (i=1; i<=steps; i++) {
        var ival = h * sum(method(f, a+i*h, h));
    }
    return ival;
}

function print(obj) {
    WScript.Echo(obj);
}

var result = integrate(f, 1, Math.PI, 100, trapezium)
print(result); 


